Source/Core/AST/CallCmd.cs

```diff
@@ -395,6 +395,15 @@ void CheckModifies(IEnumerable<Variable> modifiedVars)
           }
         }
       }
+      if(!IsAsync){
+        for(int i = 1; i <= callerDecl.Layer; i++){
```

Comment:
Isn't it better to start ```i`` from calleeDecl.Layer?

Reply:
Calling ```RefinedActionAtLayer``` might not work. I see that its implementation has strong preconditions. It might be better to implement a custom loop similar to that used in the implementation of ```RefinedActionAtLayer```.